### PR TITLE
Testing the controllers 

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of blog app we have added the following requirements : 

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controllers_spec/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      expect(response).to be_successful
+    end
+it "renders the index template" do
+  get :index, params: { user_id: 1 }
+  expect(response).to render_template("index")
+end
+
+  end
+
+end
```
Comment (julie-ify) [inline spec/controllers_spec/posts_controller_spec.rb]: - [ ] Kindly indent your tests to enhance readability and look more organized.

- [ ] Kindly write more comprehensive tests, like testing to ensure placeholder text in the post index page.

  ```
  expect(response.body).to include("Welcome to post index page")
  ```

- [ ] Kindly add tests for the post show requests.

File: spec/controllers_spec/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# spec/controllers/posts_controller_spec.rb
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
```
Comment (julie-ify) [inline spec/controllers_spec/posts_controller_spec.rb]: - [ ] Kindly note that the tests should be of `type: :request`, not `type: :controller`, as the latter is now somewhat deprecated.

File: spec/controllers_spec/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+          it "renders the index template" do
+            get :index
+            expect(response).to render_template("index")
+          end
+              end
+
+ 
+      
+  end
+
```
Comment (julie-ify) [inline spec/controllers_spec/users_controller_spec.rb]: - [ ] Kindly indent your tests to enhance readability and look more organized.

- [ ] Kindly write more comprehensive tests, like testing to ensure placeholder text in the user index page.

  ```
  expect(response.body).to include("Welcome to user index page")
  ```

- [ ] Kindly add tests for the user show requests.

File: spec/controllers_spec/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+  require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
```
Comment (julie-ify) [inline spec/controllers_spec/users_controller_spec.rb]: - [ ] Kindly refactor to use `type: :request`.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @ridabensalem,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers_spec/users_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe 'GET /users/:id' do
+    let(:user) { FactoryBot.create(:user) }
+    it 'should return a serialized user' do
+      get user_path(user)
+      expect(response).to have_http_status(200)
+    end
```
Comment (mckent05) [inline spec/controllers_spec/users_controller_spec.rb]: - Good job done so far, but as the previous reviewer suggested, your tests are not complete. According to the project requirements, each action in your controller should be testing for these:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.
Please ensure all actions are properly tested, according to the project requirements 👍 👍

File: spec/controllers_spec/posts_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    it 'works! (now write some real specs)' do
+      get posts_path
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns all posts' do
+      expect(Post.count).to eq(0)
+    end
+  end
+end
```
Comment (mckent05) [inline spec/controllers_spec/posts_controller_spec.rb]: - Good job done so far, but as the previous reviewer suggested, your tests are not complete. According to the project requirements, each action in your controller should be testing for these:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.
Please ensure all actions are properly tested, according to the project requirements 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @ridabensalem, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers_spec/posts_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    end
+
+    it 'returns all posts' do
+      expect(Post.count).to eq(0)
+    end
+
+    it 'renders the index template' do
+      get posts_path
+      expect(response).to render_template(:index)
+    end
```
Comment (codepantha) [inline spec/controllers_spec/posts_controller_spec.rb:23]: - [ ] Please write tests for the `show` action on the `PostsController`. These tests should at least do the following:
1. Ensure correct view template is rendered. E.g: `expect(response).to render_template(:show)`
2. Ensure the response body includes the correct placeholder text

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @ridabensalem 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good code organization and working app
✔️ Added controller specs 📌 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**